cluster_pack/packaging.py

```diff
@@ -406,10 +406,16 @@ def detect_archive_names(
 
 
 def resolve_zip_from_pex_dir(pex_dir: str) -> str:
-    pex_files = glob.glob(f"{os.path.dirname(pex_dir)}/*.pex.zip")
-    assert len(pex_files) == 1, \
-        f"Expected to find single zipped PEX in same dir as {pex_dir}, got {pex_files}"
-    return pex_files[0]
+    parent_dir = os.path.dirname(pex_dir)
+    pex_files = glob.glob(f"{parent_dir}/*.pex.zip")
+    if len(pex_files) == 1:
+        return pex_files[0]
+
+    pex_file = next((x for x in pex_files if pex_dir.endswith(x.split(".zip")[0])), None)

```

Comment:
Selecting the first `.pex.zip` available here, but why the first and not another?

Reply:
In fact we select the one that has the same name as pex_dir. I made the assumption it was unique, but it might not be the case actually, as I dont get an exact match, but only a `endswith`, so we could actually get false matches.

I'll fix this